presto-postgresql/src/main/java/io/prestosql/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -112,13 +119,38 @@ protected ResultSet getTables(Connection connection, Optional<String> schemaName
                 new String[] {"TABLE", "VIEW", "MATERIALIZED VIEW", "FOREIGN TABLE"});
     }
 
+    private Function<JdbcColumnHandle, String> bindingPlaceholder()
+    {
+//        return (columnHandle) -> "?::" + columnHandle.getJdbcTypeHandle().getJdbcTypeName();
+        return (columnHandle) -> "?";
+    }
+
+    @Override
+    public PreparedStatement buildSql(ConnectorSession session, Connection connection, JdbcSplit split, List<JdbcColumnHandle> columnHandles)
+            throws SQLException
+    {
+        return new QueryBuilder(identifierQuote, bindingPlaceholder()).buildSql(
+                this,
+                session,
+                connection,
+                split.getCatalogName(),
+                split.getSchemaName(),
+                split.getTableName(),
+                columnHandles,
+                split.getTupleDomain(),
+                split.getAdditionalPredicate());
+    }
+
     @Override
     public Optional<ColumnMapping> toPrestoType(ConnectorSession session, JdbcTypeHandle typeHandle)
     {
         switch (typeHandle.getJdbcTypeName()) {
             case "jsonb":
             case "json":
-                return Optional.of(jsonColumnMapping());
+                return Optional.of(jsonColumnMapping(typeHandle.getJdbcTypeName()));
+        }
+        if (typeHandle.getJdbcType() == Types.VARCHAR && !typeHandle.getJdbcTypeName().equals("varchar")) {
+            return Optional.of(objectColumnMapping(typeHandle.getJdbcTypeName()));
```

Comment:
For now add explanatory comment here:
```
// This can be e.g. an ENUM

Reply:
Once you cleanup, I will see whether we can pull "is enum" information in some better way.